addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairEasyTouchHandler.java

```diff
@@ -0,0 +1,425 @@
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.pentair.PentairBindingConstants;
+import org.openhab.binding.pentair.internal.PentairPacket;
+import org.openhab.binding.pentair.internal.PentairPacketHeatSetPoint;
+import org.openhab.binding.pentair.internal.PentairPacketStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PentairEasyTouchHandler} is responsible for implementation of the EasyTouch Controller. It will handle
+ * commands sent to a thing and implements the different channels. It also parses/disposes of the packets seen on the
+ * bus from the controller.
+ *
+ * @author Jeff James - Initial contribution
+ */
+public class PentairEasyTouchHandler extends PentairBaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(PentairEasyTouchHandler.class);
+
+    /**
+     * current/last status packet recieved, used to compare new packet values to determine if status needs to be updated
+     */
+    protected PentairPacketStatus p29cur = new PentairPacketStatus();
+    /** current/last heat set point packet, used to determine if status in framework should be updated */
+    protected PentairPacketHeatSetPoint phspcur = new PentairPacketHeatSetPoint();
+
+    public PentairEasyTouchHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing EasyTouch - Thing ID: {}.", this.getThing().getUID());
+
+        id = ((BigDecimal) getConfig().get("id")).intValue();
+
+        // make sure there are no exisitng EasyTouch controllers
+        PentairBaseBridgeHandler bh = (PentairBaseBridgeHandler) this.getBridge().getHandler();
+        List<Thing> things = bh.getThing().getThings();
+
+        for (Thing t : things) {
+            if (t.getUID().equals(this.getThing().getUID())) {
+                continue;
+            }
+            if (t.getThingTypeUID().equals(EASYTOUCH_THING_TYPE)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Another EasyTouch controller is already configured.");
+                return;
+            }
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Thing {} disposed.", getThing().getUID());
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
please do not set the status anymore in the dispose - the framework takes care of that.